pkg/scheduler/internal/cache/cache_test.go

```diff
@@ -1365,6 +1418,7 @@ func TestSchedulerCache_UpdateNodeInfoSnapshot(t *testing.T) {
 }
 
 func compareCacheWithNodeInfoSnapshot(cache *schedulerCache, snapshot *nodeinfosnapshot.Snapshot) error {
```

Comment:
In addition to comparing cache with snapshot, I think we should also make sure the cache itself if self consistent, i.e., `cache.nodes`, `cache.headNode` and `cache.nodeTree` should contain the same nodes.

Not sure if this is already covered by other tests.

Reply:
But that is not part of the UpdateNodeInfoSnapshot, if not covered should certainly be done, but not in this PR.

Reply:
Yeah agreed.